packages/teleterm/src/ui/TabHost/ClusterConnectPanel/RecentClusters.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import { getClusterName, getUserWithClusterName } from 'teleterm/ui/utils';
+import { Box, ButtonBorder, Card, Text } from 'design';
+
+export function RecentClusters() {
+  const ctx = useAppContext();
+
+  ctx.clustersService.useState();
+
+  const rootClusters = ctx.clustersService
+    .getClusters()
+    .filter(c => !c.leaf)
+    .map(cluster => ({
+      userWithClusterName: getUserWithClusterName({
+        userName: cluster.loggedInUser?.name,
+        clusterName: getClusterName(cluster),
+      }),
+      uri: cluster.uri,
+    }));
+
+  function connect(clusterUri: string): void {
+    ctx.workspacesService.setActiveWorkspace(clusterUri);
+  }
+
+  if (!rootClusters.length) {
+    return null;
+  }
+
+  return (
+    <Card py={3} maxWidth="480px" mx={4} mb={4} bg="primary.main">
```

Comment:
The width of this still doesn't seem to be right. The card still doesn't take all available space if the text by itself doesn't make the card wide enough.

![smol card](https://user-images.githubusercontent.com/27113/172411606-4ffc728e-3e8a-4b61-b2f6-de55cf80dc91.png)

After I added `width: 100%` to it, I noticed that it's not aligned properly, the flex column above Card needs `align-items: center`. And the card still grew only to 370px for some reason but I think we can just give up on this specific issue, unless you have an idea on what's happening.